selfdrive/test/process_replay/process_replay.py

```diff
@@ -816,3 +816,22 @@ def check_openpilot_enabled(msgs: LogIterable) -> bool:
       max_enabled_count = max(max_enabled_count, cur_enabled_count)
 
   return max_enabled_count > int(10. / DT_CTRL)
+
+
+def check_most_messages_valid(msgs: LogIterable, threshold: float = 0.9) -> bool:
+  msgs_counts: dict[str, int] = defaultdict(int)
+  msgs_valid_counts: dict[str, int] = defaultdict(int)
+  for msg in msgs:
+    msgs_counts[msg.which()] += 1
+    if msg.valid:
+      msgs_valid_counts[msg.which()] += 1
+
+  most_valid_for_service = {}
+  for msg_type in msgs_counts.keys():
+    valid_share = msgs_valid_counts[msg_type] / msgs_counts[msg_type]
```

Comment:
```suggestion
    valid_share = msgs_valid_counts.get(msg_type, 0) / msgs_counts[msg_type]
```

in case a process doesn't output any valid msgs